BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,299 @@
+    "arm64_cmake_prefix":  '/opt/homebrew',
+    "x86_64_cmake_prefix": '/usr/local',
+
+    # Locations to qt5 directories for arm and x64 libraries
+    # The default values of these paths are taken from the default
+    # paths used for homebrew
+    "arm64_qt5_path":  '/opt/homebrew/opt/qt5',
+    "x86_64_qt5_path": '/usr/local/opt/qt5',
+
+    # Identity to use for code signing. "-" indicates that the app will not
+    # be cryptographically signed/notarized but will instead just use a
+    # SHA checksum to verify the integrity of the app. This doesn't
+    # protect against malicious actors, but it does protect against
+    # running corrupted binaries and allows for access to the extended
+    # permisions needed for ARM builds
+    "codesign_identity":  '-',
+    # Entitlements file to use for code signing
+    "entitlements": "../Source/Core/DolphinQt/DolphinEmu.entitlements",
+
+    # Minimum macOS version for each architecture slice
+    "arm64_mac_os_deployment_target":  "11.0.0",
+    "x86_64_mac_os_deployment_target": "10.12.0",
+
+    # CMake Generator to use for building
+    "generator": "Unix Makefiles",
+    "build_type": "Release"
+
+}
+
+# Architectures to build for. This is explicity left out of the command line
+# config options for several reasons:
+# 1) Adding new architectures will generally require more code changes
+# 2) Single architecture builds should utilize the normal generated cmake
+#    project files rather than this wrapper script
+
+ARCHITECTURES = ["x86_64", "arm64"]
+
+
+def parse_args(conf=DEFAULT_CONFIG):
+    """
+    Parses the command line arguments into a config dictionary.
+    """
+
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument(
+        '--target',
+        help='Build target in generated project files',
+        default=conf["build_target"],
+        dest="build_target")
+    parser.add_argument(
+        '-G',
+        help='CMake Generator to use for creating project files',
+        default=conf["generator"],
+        dest="generator")
+    parser.add_argument(
+        '--build_type',
+        help='CMake build type [Debug, Release, RelWithDebInfo, MinSizeRel]',
+        default=conf["build_type"],
+        dest="build_type")
+    parser.add_argument(
+        '--dst_app',
+        help='Directory where universal binary will be stored',
+        default=conf["dst_app"])
+
+    parser.add_argument(
+        '--entitlements',
+        help='Path to .entitlements file for code signing',
+        default=conf["entitlements"])
+
+    parser.add_argument(
+        '--codesign',
+        help='Code signing identity to use to sign the applications',
+        default=conf["codesign_identity"],
+        dest="codesign_identity")
+
+    for arch in ARCHITECTURES:
+        parser.add_argument(
+             '--{}_cmake_prefix'.format(arch),
+             help="Folder for cmake to search for packages".format(arch),
```

Comment:
```suggestion
             help="Folder for cmake to search for packages",
```
(I guess having too many format arguments isn't an error...)